lib/wind/utf8.c

```diff
@@ -412,15 +429,30 @@ wind_utf8ucs2(const char *in, uint16_t *out, size_t *out_len)
 	if (ret)
 	    return ret;
 
-	if (u & 0xffff0000)
-	    return WIND_ERR_NOT_UTF16;
```

Comment:
We should have an error code `WIND_ERR_NOT_BMP` or something.

Reply:
This was the only place where we used `WIND_ERR_NOT_UTF16`, so maybe we should rename it to `WIND_ERR_NOT_BMP`.

Reply:
We already have `WIND_ERR_INVALID_UTF16` for use in decoding of UTF-16, so we really don't need `WIND_ERR_NOT_UTF16`, so let's rename this to `WIND_ERR_NOT_BMP`, and let's use it below.